Signal/src/AppDelegate.m

```diff
@@ -239,7 +239,11 @@ - (void)applicationDidBecomeActive:(UIApplication *)application {
                                                [TSSocketManager becomeActiveFromForeground];
                                                [[Environment getCurrent].contactsManager verifyABPermission];
                                            }];
-
+    
+    if ([TSAccountManager isRegistered]) {
+        [[PushManager sharedManager] applicationDidBecomeActive];
```

Comment:
`[TSAccountManager isRegistered]` **can** be expensive.

It reads from the DB, so it has to wait for the DB to be ready before it can run. DB setup is normally not expensive, unless the user is launching for the first time after a new release which adds a database extension like a new view. Depending on the size of the users database, populating a new extension can be very expensive.